lib/Remotery.c

```diff
@@ -167,6 +167,10 @@ static rmtBool g_SettingsInitialized = RMT_FALSE;
     #include <cuda.h>
 #endif
 
+#if defined(RMT_USE_C_ATOMICS)
+    #include <stdatomic.h>
+#endif
+
```

Comment:
I removed the compiler checks for this define, as it seems like both linux (gcc) and windows (cl) doesn't have it implemented?

Reply:
Looking further into the build error on [linux/gcc](https://github.com/JCash/Remotery/actions/runs/3043210148/jobs/4902226495):
`error: ‘_Atomic’ does not name a type`

One suggested way to solve it for C++ would be:
```
#ifndef __cplusplus
# include <stdatomic.h>
#else
# include <atomic>
# define _Atomic(X) std::atomic< X >
#endif
```

Not sure how I feel about it, but I think we'd need something like this for compilers that implement stdatomic.h this way.

Benefit is that then the `RMT_USE_C_ATOMICS` can be set using the C11/C++11 check again.

Reply:
This is totally not how I would do it, as I would have added separate `RMT_USE_C11_ATOMICS` and `RMT_USE_CPP_ATOMICS`.

But I have to admit: this way saves adding a whole bunch of C++ cases. And it actually seems to work.

For now, I think this is good enough and if we need to make it more succinct in future, we can easily do so.

Reply:
Yeah, and it didn't really work cleanly either.
So the latest version is playing with these two different defines.